src/coral/docs/directory-structure.md

```diff
@@ -0,0 +1,134 @@
+# Directory structure
+
+The structure is inspired in big parts by:
+
+- [How To Structure React Projects From Beginner To Advanced](https://blog.webdevsimplified.com/2022-07/react-folder-structure/). See the "Advanced Folder Structure".
+- [Elegant Frontend Architecture](https://michalzalecki.com/elegant-frontend-architecture/).
+
+## 🏁️ Primary goals of this structure
+
+- We want a strong separation between the frontend app and the backend language.
+  Having a `domain` directory on top level enables us to do so. Nothing inside of `app` should have any knowledge about anything related to backend. Example: data model from incoming API data.
+
+- We want a comprehensive approach, which enables us to move fast and the community to contribute.
+  Choosing an approach based on documented, established pattern avoids re-inventing the wheel. It also enables us to provide a lot of background references for the community.
+
+- We want to avoid spending too much time on "Which file is where?"
+  By having only a `services` directory and not additional `lib` (or `utils`), we reduce the brain load for this.
+
+## 🗂️ Folder and file naming
+
+- folders are in `kebab-case`
+- TypeScript files are in `kebab-case`
```

Comment:
My personal preference is kebab-cases, but I'm fine with either 😅 linting will make sure everything is consistent either way (and consistency is the most important part here)!

I thought maybe we even go in using only kebab-cases (or snake_case) based on [Git is case-sensitive and your filesystem may not be - Weird folder merging on Windows](https://www.hanselman.com/blog/git-is-casesensitive-and-your-filesystem-may-not-be-weird-folder-merging-on-windows) 🤔  This would make sure that we don't run into issues as described when contributor are using different machines. Directory and file ending would already convey the information what kind of code the file contains (react vs typescript). That's the only technical argument I found for file naming convention.

Reply:
Using all kebab cases or only typescript files? In first case I would update the doc accordingly.